mp/src/materialsystem/stdshaders/fxctmp9/neo_thermal_model_vs20.inc

```diff
@@ -0,0 +1,160 @@
+#include "shaderlib/cshader.h"
+class neo_thermal_model_vs20_Static_Index
+{
+public:
+	neo_thermal_model_vs20_Static_Index( )
+	{
+	}
+	int GetIndex()
+	{
+		// Asserts to make sure that we aren't using any skipped combinations.
+		// Asserts to make sure that we are setting all of the combination vars.
+#ifdef _DEBUG
+#endif // _DEBUG
+		return 0;
+	}
+};
+#define shaderStaticTest_neo_thermal_model_vs20 0
+class neo_thermal_model_vs20_Dynamic_Index
+{
+private:
+	int m_nCOMPRESSED_VERTS;
+#ifdef _DEBUG
+	bool m_bCOMPRESSED_VERTS;
+#endif
+public:
+	void SetCOMPRESSED_VERTS( int i )
+	{
+		Assert( i >= 0 && i <= 1 );
+		m_nCOMPRESSED_VERTS = i;
+#ifdef _DEBUG
+		m_bCOMPRESSED_VERTS = true;
+#endif
+	}
+	void SetCOMPRESSED_VERTS( bool i )
+	{
+		m_nCOMPRESSED_VERTS = i ? 1 : 0;
+#ifdef _DEBUG
+		m_bCOMPRESSED_VERTS = true;
+#endif
+	}
+private:
+	int m_nDOWATERFOG;
+#ifdef _DEBUG
+	bool m_bDOWATERFOG;
```

Comment:
What's the reason for having the underlying type be different based on debug/release build mode? As any nonzero value evaluates as `true` for `bool`, it's probably safe to just use `bool` for both modes, and do the assertion sanity checks in debug for the input value. Unless there's some other consideration in play here.

Reply:
ditto on example model include file